Objective-C/CBLDatabase.mm

```diff
@@ -823,7 +823,9 @@ - (BOOL) open: (NSError**)outError {
     if (!_c4db)
         return convertError(err, outError);
     
-    _sharedKeys = c4db_getFLSharedKeys(_c4db);
+    CBL_LOCK(self) {
+        _sharedKeys = c4db_getFLSharedKeys(_c4db);
+    }
```

Comment:
I think we don't need to lock this as the open is called when CBLDatabase object is created so there shouldn't be anything concurrent yet.

Reply:
`initWithName: config: error` is a public API right, which can be called concurrently

Reply:
nvm, each call will end up in separate db instances.